add pipeline promotion documentation

makkes opened the pull request:
Preview: https://staging.docs.gitops.weave.works/github-pr-promotion-docs/docs/next/enterprise/pipelines/github-pr-promotions/

closes weaveworks/pipeline-controller#49

makkes commented:
> Although I've noticed one thing, it would be nice to rename that file from `github-pr-promotions.mdx` to `promotions.mdx` and make the page outline that we have multiple strategies, so I can on top of that add the notification one to the list.

Shouldn't each strategy be documented on its own page?

sympatheticmoose commented on website/docs/enterprise/pipelines/github-pr-promotions.mdx:
```diff
@@ -0,0 +1,121 @@
+
+# Promoting apps through pipeline environments <TierLabel tiers="enterprise" />
+
+Pipelines allow you to configure automatic promotions of applications through a consecutive set of environments, e.g. from dev to staging to production. The environments are defined in the `Pipeline` resource itself so that each pipeline governs a single application and all the environments it is deployed to.
```
Perhaps we should explicitly call out "via pull request" if this will remain scoped to the pr strategy?

sympatheticmoose commented on website/docs/enterprise/pipelines/github-pr-promotions.mdx:
```diff
@@ -0,0 +1,121 @@
+
+In order to get started with promotions you start by defining the app's pipeline along with all the environments as described in the [Getting started guide](../getting-started). Whenever the app defined in a pipeline is upgraded one of the the pipeline's environments, a GitHub PR is created that updates the manifest file setting the app version in the next environment.
+
+The dynamic nature of GitOps deployments requires you to assist Weave GitOps a little with information on which repository hosts the manifest files, how to authenticate with the repository and the GitHub API and which file hosts the version definition for each environment.
```
we need to detail requirements on the token

sympatheticmoose commented on website/docs/enterprise/pipelines/github-pr-promotions.mdx:
```diff
@@ -0,0 +1,121 @@
+
+## Create credentials secret
+
+In order to authenticate with the GitHub repository hosting the manifests, you need to create a Secret with credentials used for cloning. The contents of this secret are the same you use for [GitRepositories](https://fluxcd.io/flux/components/source/gitrepositories/#secret-reference). In addition to these you need to include a `token` field containing a GitHub access token that is used to authent ...
```
we need to detail requirements for the token

makkes commented:
@sympatheticmoose addressed all of your comments except

> view of pipeline details in WGE

as I think such a screenshot is already on http://docs.gitops.weave.works/docs/enterprise/pipelines/getting-started/#view-the-details-of-a-pipeline
